src/System.Private.CoreLib/shared/System/Globalization/CultureInfo.cs

```diff
@@ -1296,15 +1192,44 @@ public static CultureInfo GetCultureInfo(string name, string altName)
                 throw new ArgumentNullException(nameof(altName));
             }
 
-            CultureInfo? retval = GetCultureInfoHelper(-1, name, altName);
-            if (retval == null)
+            name = CultureData.AnsiToLower(name);
+            altName = CultureData.AnsiToLower(altName);
+            string nameAndAltName = name + "\xfffd" + altName;
+            Dictionary<string, CultureInfo> nameTable = CachedCulturesByName;
```

Comment:
>CachedCulturesByName [](start = 56, length = 20)

nit: maybe we can use a different Dictionary for this scenario. This is not common scenario and it has its own keys notation anyway.